sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -118,10 +163,18 @@ def fit(self, X, y=None):
         for x in X:
             for f, v in x.items():
                 if isinstance(v, str):
-                    f = "%s%s%s" % (f, self.separator, v)
-                if f not in vocab:
-                    feature_names.append(f)
-                    vocab[f] = len(vocab)
+                    feature_name = "%s%s%s" % (f, self.separator, v)
+                    v = 1
+                elif isinstance(v, Number) or (v is None):
+                    feature_name = f
+                elif isinstance(v, Iterable):
```

Comment:
we should probably still make sure this is not a Mapping

Reply:
Done, but not sure this is the best way of doing it.